playwright/tests/pub_yearFilter.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { Given, Then, When } from "@cucumber/cucumber";
+import { expect } from "@playwright/test";
+import { CustomWorld } from "../support/setup";
+
+Given(
+  "I want to filter by publication year",
+  async function (this: CustomWorld) {
+    return await this.page.goto(`${this.parameters.appUrl}`);
+  }
+);
+
+When(
+  "I search for New York to test publication year filter",
+  async function (this: CustomWorld) {
+    await this.page.locator('[aria-label="Item Search"]').fill("New York");
+    await this.page.locator("#searchbar-button-search-bar").click();
+  }
+);
+
+Then("I input publication years 1900-1900", async function (this: CustomWorld) {
+  await this.page
+    .locator('[aria-describedby="date-filter-from-helperText"]')
```

Comment:
Let's use the name attribute since it's more human readable. `[name="Date From"]`

Reply:
done.